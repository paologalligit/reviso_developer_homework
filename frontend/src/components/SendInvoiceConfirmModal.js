/* eslint-disable react/prop-types */
import React, { Component } from 'react';
import { Button, Modal } from 'semantic-ui-react';

class SendInvoiceConfirmModal extends Component {
  state = { open: false };

  closeConfigShow = (closeOnEscape, closeOnDimmerClick) => () => {
    this.setState({ closeOnEscape, closeOnDimmerClick, open: true });
  };

  close = () => this.setState({ open: false });

  render() {
    const { open, closeOnEscape, closeOnDimmerClick } = this.state;
    const { onClick, label, disabled } = this.props;

    return (
      <div>
        <Button
          onClick={this.closeConfigShow(false, false)}
          disabled={disabled}
          primary
        >
          {label}
        </Button>

        <Modal
          open={open}
          closeOnEscape={closeOnEscape}
          closeOnDimmerClick={closeOnDimmerClick}
          onClose={this.close}
          size="mini"
        >
          <Modal.Header>Sending Invoice</Modal.Header>
          <Modal.Content>
            <p>Are you sure you want to send this invoice?</p>
          </Modal.Content>
          <Modal.Actions>
            <Button onClick={this.close} negative>
              No
            </Button>
            <Button
              onClick={() => {
                onClick();
                this.close();
              }}
              positive
              labelPosition="right"
              icon="checkmark"
              content="Yes"
            />
          </Modal.Actions>
        </Modal>
      </div>
    );
  }
}

export default SendInvoiceConfirmModal;
