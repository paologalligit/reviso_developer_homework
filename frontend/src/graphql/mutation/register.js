import gql from 'graphql-tag';

export default gql`
  mutation(
    $name: String!
    $surname: String!
    $username: String!
    $email: String!
    $password: String!
    $birthdate: String!
    $country: String!
    $city: String!
    $address: String!
    $postal: Int!
  ) {
    registerUser(
      name: $name
      surname: $surname
      username: $username
      email: $email
      password: $password
      birthdate: $birthdate
      country: $country
      city: $city
      address: $address
      postal: $postal
    ) {
      ok
      errors {
        path
        message
      }
    }
  }
`;
