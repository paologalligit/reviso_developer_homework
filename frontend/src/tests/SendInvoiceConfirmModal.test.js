/* eslint-disable func-names */
import 'jsdom-global/register';
import React from 'react';
import toJson from 'enzyme-to-json';
import { mount, render } from 'enzyme';
import { MockedProvider } from 'react-apollo/test-utils';

import SendInvoiceConfirmModal from '../containers/SendInvoiceConfirmModal';

describe('project list item suite', () => {
  it('renders correctly', () => {
    const wrapper = render(
      <MockedProvider mocks={[]}>
        <SendInvoiceConfirmModal />
      </MockedProvider>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(wrapper).toHaveLength(1);
  });
});

describe('sicm interaction', () => {
  let wrapper;
  const onClickMock = jest.fn();

  beforeEach(() => {
    wrapper = mount(
      <MockedProvider mocks={[]} addTypename={false}>
        <SendInvoiceConfirmModal onClick={onClickMock} label="Send Invoice" disabled={false} />
      </MockedProvider>,
    );
  });

  it('shows the send invoice modal if the invoice is not sent', () => {
    const sendInvoiceConfirmModal = wrapper.find('SendInvoiceConfirmModal');
    const openModalButton = wrapper.find('Button');
    expect(sendInvoiceConfirmModal.instance().state.open).toBeFalsy();
    openModalButton.simulate('click');
    expect(sendInvoiceConfirmModal.instance().state.open).toBeTruthy();
  });

  it('correctly closes if the no button is pressed', () => {
    const sendInvoiceConfirmModal = wrapper.find('SendInvoiceConfirmModal');
    const openModalButton = wrapper.find('Button');

    openModalButton.simulate('click');

    const closeButton = wrapper.find('Button').at(1);
    closeButton.simulate('click');
    expect(sendInvoiceConfirmModal.instance().state.open).toBeFalsy();
  });

  it('correctly send invoice if the yes button is pressed', () => {
    const sendInvoiceConfirmModal = wrapper.find('SendInvoiceConfirmModal');
    const openModalButton = wrapper.find('Button');

    openModalButton.simulate('click');

    const closeButton = wrapper.find('Button').at(2);
    closeButton.simulate('click');
    expect(sendInvoiceConfirmModal.instance().state.open).toBeFalsy();
    expect(onClickMock).toHaveBeenCalledTimes(1);
  });
});

/*
const mock = [
    {
      request: {
        query: sendInvoiceMutation,
        variables: {
          id: 1,
          sent: false,
          user_id: 1,
          customer_id: 1,
        },
      },
      result: {
        data: {
          sentInvoice: {
            ok: true,
            collaboration: {
              sent: true,
              user_id: 1,
              customer_id: 1,
            },
            errors: null,
          },
        },
      },
    },
  ];

*/
