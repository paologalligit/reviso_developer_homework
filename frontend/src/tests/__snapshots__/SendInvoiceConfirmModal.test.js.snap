// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`project list item suite renders correctly 1`] = `
<div>
  <button
    class="ui primary button"
  />
</div>
`;
